BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -20,13 +20,44 @@ public BlogDataStore(IFileSystem fileSystem)
         {
             _fileSystem = fileSystem;
             InitStorageFolder();
+            InitCurrentId();
         }
         public void InitStorageFolder()
         {
             _fileSystem.CreateDirectory(StorageFolder);
             _fileSystem.CreateDirectory(DraftsFolder);
         }
 
+        public void InitCurrentId()
+        {
+            int max = 0;
+            List<string> postfiles = _fileSystem.EnumerateFiles($"{StorageFolder}").ToList();

```

Comment:
Also, since all you want is the file names, not the full path, you could call `EnumerateFiles(...).Select(f => Path.GetFileName(f))` to get just the file names.  This way your parsing later won't be affected by any characters in the full path (. or _ for example could be used in a folder name)